File: tests/fixtures/python/RequestBodyPydanticModel.py
from __future__ import annotations

from pydantic import BaseModel, Field


class OwnerController:

    # @PostMapping("/owners")
    def process_creation_form(self, form: OwnerRequest) -> OwnerRequest:
        return form


class OwnerRequest(BaseModel):
    first_name: str = Field(..., min_length=1, max_length=30)
    last_name: str | None = None

File: tests/fixtures/java/RequestBodyPydanticModel.java
import jakarta.validation.Valid;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;

class OwnerController {
    @PostMapping("/owners")
    public OwnerRequest processCreationForm(@Valid @RequestBody OwnerRequest form) {
        return form;
    }
}

class OwnerRequest {
    @NotNull
    @Size(min = 1, max = 30)
    private String firstName;
    private String lastName;

    public OwnerRequest() {}
    public String getFirstName() { return firstName; }
    public void setFirstName(String firstName) { this.firstName = firstName; }
    public String getLastName() { return lastName; }
    public void setLastName(String lastName) { this.lastName = lastName; }
}
